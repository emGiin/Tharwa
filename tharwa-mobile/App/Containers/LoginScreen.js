import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Alert, View, Image, ActivityIndicator } from 'react-native'
import { Container, Content, Text } from 'native-base';
import { connect } from 'react-redux'
import PopupDialog, {
  DialogTitle,
  DialogButton,
  SlideAnimation
} from 'react-native-popup-dialog';
import { Images } from '../Themes'
import AuthActions from '../Redux/AuthRedux'
import LoginForm from '../Components/LoginForm'

// Styles
import styles from './Styles/LoginScreenStyle'

const slideAnimation = new SlideAnimation({ slideFrom: 'bottom' });

const Logo = () => (
  <View style={styles.logoContainer}>
    <Image source={Images.logo} style={styles.logo} />
  </View>
)

class LoginScreen extends Component {
  static propTypes = {
    dispatch: PropTypes.func,
    fetching: PropTypes.bool,
    error: PropTypes.string,
    success: PropTypes.bool,
    attemptLogin: PropTypes.func
  }

  componentWillReceiveProps(props) {
    if (!props.fetching) {
      if (props.error && props.error === 'WRONG') {
        // Alert.alert('Erreur', 'Email ou mot de passe incorrect!', [{ text: 'Fermer' }])
        this.slideAnimationDialog.dismiss();
      } else if (props.success && !this.props.success) {
        this.slideAnimationDialog.dismiss();
        this.goToPinCodePage();
      }
    }
  }

  submit = (confirmationMethod) => {
    this.props.attemptLogin(this.email, this.password, confirmationMethod)
  }

  loginFormSubmit = (values) => {
    this.email = values.email;
    this.password = values.password;
    this.slideAnimationDialog.show();
  }

  goToPinCodePage = () => {
    this.props.navigation.navigate('PinCodeScreen');
  }

  goToSignUpPage = () => {
    this.props.navigation.navigate('RegisterScreen');
  }

  render() {
    const { fetching, error } = this.props;
    return (
      <Container style={{ backgroundColor: '#2c3e50' }}>
        <PopupDialog
          width={0.95}
          height={170}
          ref={(popupDialog) => { this.slideAnimationDialog = popupDialog; }}
          dialogAnimation={slideAnimation}
          dialogTitle={<DialogTitle title="Code d'authentification" />}>
          <View style={styles.dialogContentView}>
            <Text style={styles.dialogContent}>
              Comment souhaiter-vous recevoir votre code d'authentification?
            </Text>
            {
              fetching ? <ActivityIndicator size='large' />
                : <View style={{ flex: 1, flexDirection: 'row' }}>
                  <DialogButton disabled={fetching} text="Par Email" key="button-1"
                    onPress={() => { this.submit('email'); }} />
                  <DialogButton disabled={fetching} text="Par SMS" key="button-2"
                    onPress={() => { this.submit('sms'); }} />
                </View>
            }
          </View>
        </PopupDialog>
        <Content>
          <Logo />
          <LoginForm
            onSubmit={this.loginFormSubmit}
            onRegisterClicked={this.goToSignUpPage}
            editable={!fetching}
          />
          {!!error && <Text>{error}</Text>}
        </Content>
      </Container>
    )
  }
}

const mapStateToProps = (state) => {
  const { fetching, error, success } = state.auth;
  return { fetching, error, success }
}

const mapDispatchToProps = (dispatch) => {
  return {
    attemptLogin: (email, password, confirmationMethod) =>
      dispatch(AuthActions.authRequest(email, password, confirmationMethod)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginScreen)
