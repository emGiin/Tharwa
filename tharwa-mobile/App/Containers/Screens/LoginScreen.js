import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, Image, ActivityIndicator } from 'react-native'
import { Container, Content, Text } from 'native-base'
import { connect } from 'react-redux'
import PopupDialog, {
  DialogTitle,
  DialogButton,
  SlideAnimation
} from 'react-native-popup-dialog'
import I18n from 'react-native-i18n'
import { Images } from '../../Themes'
import AuthActions from '../../Redux/AuthRedux'
import LoginForm from '../Forms/LoginForm'

// Styles
import styles from './Styles/LoginScreenStyle'

const slideAnimation = new SlideAnimation({ slideFrom: 'bottom' });

class LoginScreen extends Component {
  static propTypes = {
    fetching: PropTypes.bool,
    error: PropTypes.string,
    success: PropTypes.bool,
    attemptLogin: PropTypes.func
  }

  constructor(props) {
    super(props)
    this.state = { newRequest: true }
  }

  componentWillReceiveProps(props) {
    if (!props.fetching && props.success) {
      this.dialog.dismiss();
      this.goToPinCodePage();
    }
    if (props.error) this.setState({ newRequest: false })
  }

  submit = (confirmationMethod) => {
    this.props.attemptLogin(this.email, this.password, confirmationMethod)
  }

  loginFormSubmit = (values) => {
    this.email = values.email;
    this.password = values.password;
    this.setState({ newRequest: true })
    this.dialog.show();
  }

  goToPinCodePage = () => {
    this.props.navigation.navigate('PinCodeScreen');
  }

  goToSignUpPage = () => {
    this.props.navigation.navigate('RegisterScreen');
  }

  render() {
    const { fetching, error } = this.props;
    let dialogTitle = I18n.t('authDialogTitle');
    let dialogDescription = I18n.t('authDialogDescription');
    let dialogContent;
    if (fetching) {
      dialogTitle = I18n.t('authDialogTitleFetching');
      dialogDescription = I18n.t('authDialogDescriptionFetching');
      dialogContent = <ActivityIndicator size='large' style={{ marginVertical: 20 }} />;
    } else if (error && !this.state.newRequest) {
      dialogTitle = I18n.t('authDialogTitleError')
      dialogDescription = error
      dialogContent = <DialogButton text={I18n.t('authDialogClose')}
        onPress={() => { this.dialog.dismiss(); }} />
    } else {
      dialogContent =
        <View style={{ flex: 1, flexDirection: 'row' }}>
          <DialogButton disabled={fetching} text={I18n.t('authDialogPinByEmail')}
            onPress={() => { this.submit('email'); }} />
          <DialogButton disabled={fetching} text={I18n.t('authDialogPinBySms')}
            onPress={() => { this.submit('sms'); }} />
        </View>
    }
    return (
      <Container style={{ backgroundColor: '#2c3e50' }}>
        <PopupDialog
          width={0.95}
          height={170}
          dismissOnTouchOutside={!fetching}
          dismissOnHardwareBackPress={!fetching}
          ref={(dialog) => { this.dialog = dialog; }}
          dialogAnimation={slideAnimation}
          dialogTitle={<DialogTitle title={dialogTitle} />}
        >
          <View style={styles.dialogContentView}>
            <Text style={styles.dialogContent}>
              {dialogDescription}
            </Text>
            {dialogContent}
          </View>
        </PopupDialog>
        <Content>
          <View style={styles.logoContainer}>
            <Image source={Images.logo} style={styles.logo} />
          </View>
          <LoginForm
            onSubmit={this.loginFormSubmit}
            onRegisterClicked={this.goToSignUpPage}
            editable={!fetching}
          />
        </Content>
      </Container>
    )
  }
}

const mapStateToProps = (state) => {
  const { fetching, error, success } = state.auth;
  return { fetching, error, success }
}

const mapDispatchToProps = (dispatch) => {
  return {
    attemptLogin: (email, password, confirmationMethod) =>
      dispatch(AuthActions.authRequest(email, password, confirmationMethod)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginScreen)
