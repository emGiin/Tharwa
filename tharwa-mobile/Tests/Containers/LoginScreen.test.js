import 'react-native'
import React from 'react'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import AuthActions from '../../App/Redux/AuthRedux'
import LoginScreen from '../../App/Containers/LoginScreen'

describe('Login Screen container', () => {
  const initialState = {
    auth: {
      fetching: false, error: null, success: false
    }
  }
  const mockStore = configureStore()
  const dispatchSpy = jest.fn();
  const navigationSpy = jest.fn();
  let store, container, content;

  beforeAll(() => {
    store = mockStore(initialState);
    store.dispatch = dispatchSpy;
    container = shallow(
      <LoginScreen navigation={{ navigate: navigationSpy }} />,
      { context: { store } }
    )
    content = container.dive()
  })

  it('should check authentication response', () => {
    const instance = content.instance();
    const spy = jest.fn();
    instance.dialog = { dismiss: spy };
    content.setProps({ error: 'WRONG' });
    instance.componentWillReceiveProps({ fetching: false, error: 'WRONG' });
    expect(content.find('Styled(Text)')).toHaveLength(1)

    instance.componentWillReceiveProps({ fetching: false, success: true });
    expect(spy).toHaveBeenCalled()
    expect(navigationSpy).toHaveBeenCalled()
    navigationSpy.mockClear()
  })

  it('should render the component', () => {
    expect(container.length).toEqual(1)
  });

  it('should check Prop matching with initialState', () => {
    const { auth } = initialState;
    expect(container.prop('fetching')).toEqual(auth.fetching)
    expect(container.prop('error')).toEqual(auth.error)
    expect(container.prop('success')).toEqual(auth.success)
  });

  it('should check container initial values of email and password', () => {
    const instance = container.instance();
    expect(instance.email).toBeUndefined()
    expect(instance.password).toBeUndefined()
  });

  it('should call the login function', () => {
    const instance = content.instance();
    const showSpy = jest.fn();
    instance.dialog = { show: showSpy }
    content.find('ReduxForm').simulate('submit', {
      email: 'user@email.com', password: 'password'
    })
    expect(instance.email).toBe('user@email.com')
    expect(instance.password).toBe('password')
    expect(showSpy).toHaveBeenCalled()
  })

  it('should call the submit function', () => {
    content.find('DialogButton').at(0).simulate('press');
    expect(dispatchSpy).toHaveBeenCalled();
    expect(dispatchSpy).toHaveBeenCalledWith(AuthActions.authRequest(
      'user@email.com', 'password', 'email'
    ));
    dispatchSpy.mockClear()
    content.find('DialogButton').at(1).simulate('press');
    expect(dispatchSpy).toHaveBeenCalled();
    expect(dispatchSpy).toHaveBeenCalledWith(AuthActions.authRequest(
      'user@email.com', 'password', 'sms'
    ));
    dispatchSpy.mockClear()
  })

  it('should show activity indicator when fetching', () => {
    content.setProps({ fetching: true });
    expect(content.find('ActivityIndicator')).toHaveLength(1)
  })

  it('should call the submit function', () => {
    const instance = content.instance();
    instance.goToPinCodePage();
    expect(navigationSpy).toHaveBeenCalled();
    expect(navigationSpy).toHaveBeenCalledWith('PinCodeScreen');
    navigationSpy.mockClear()
    instance.goToSignUpPage();
    expect(navigationSpy).toHaveBeenCalled();
    expect(navigationSpy).toHaveBeenCalledWith('RegisterScreen');
    navigationSpy.mockClear()
  })
});