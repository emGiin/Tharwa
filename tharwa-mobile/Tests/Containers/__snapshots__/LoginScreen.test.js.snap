// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen container should render as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginScreen
    attemptLogin={[Function]}
    error={null}
    fetching={false}
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
    success={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <PopupDialog
          dialogAnimation={
            SlideAnimation {
              "animate": 0,
              "animations": Object {
                "transform": Array [
                  Object {
                    "translateY": 800,
                  },
                ],
              },
            }
          }
          dialogTitle={
            <DialogTitle
              haveTitleBar={true}
              title="Code d'authentification"
              titleAlign="center"
            />
          }
          height={170}
          width={0.95}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Styled(Text)
              style={
                Object {
                  "color": "#555",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
              virtual={undefined}
            >
              Comment souhaiter-vous recevoir votre code d'authentification?
            </Styled(Text)>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <DialogButton
                align="center"
                disabled={false}
                onPress={[Function]}
                text="Par Email"
              />
              <DialogButton
                align="center"
                disabled={false}
                onPress={[Function]}
                text="Par SMS"
              />
            </View>
          </View>
        </PopupDialog>,
        <Styled(Content)
          virtual={undefined}
        >
          <Logo />
          <ReduxForm
            editable={true}
            onRegisterClicked={[Function]}
            onSubmit={[Function]}
          />
        </Styled(Content)>,
      ],
      "style": Object {
        "backgroundColor": "#2c3e50",
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Styled(Text)
              style={
                Object {
                  "color": "#555",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
              virtual={undefined}
            >
              Comment souhaiter-vous recevoir votre code d'authentification?
            </Styled(Text)>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <DialogButton
                align="center"
                disabled={false}
                onPress={[Function]}
                text="Par Email"
              />
              <DialogButton
                align="center"
                disabled={false}
                onPress={[Function]}
                text="Par SMS"
              />
            </View>
          </View>,
          "dialogAnimation": SlideAnimation {
            "animate": 0,
            "animations": Object {
              "transform": Array [
                Object {
                  "translateY": 800,
                },
              ],
            },
          },
          "dialogTitle": <DialogTitle
            haveTitleBar={true}
            title="Code d'authentification"
            titleAlign="center"
          />,
          "height": 170,
          "width": 0.95,
        },
        "ref": [Function],
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Text)
                style={
                  Object {
                    "color": "#555",
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
                virtual={undefined}
              >
                Comment souhaiter-vous recevoir votre code d'authentification?
              </Styled(Text)>,
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <DialogButton
                  align="center"
                  disabled={false}
                  onPress={[Function]}
                  text="Par Email"
                />
                <DialogButton
                  align="center"
                  disabled={false}
                  onPress={[Function]}
                  text="Par SMS"
                />
              </View>,
            ],
            "style": Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Comment souhaiter-vous recevoir votre code d'authentification?",
                "style": Object {
                  "color": "#555",
                  "fontSize": 20,
                  "textAlign": "center",
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Comment souhaiter-vous recevoir votre code d'authentification?",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <DialogButton
                    align="center"
                    disabled={false}
                    onPress={[Function]}
                    text="Par Email"
                  />,
                  <DialogButton
                    align="center"
                    disabled={false}
                    onPress={[Function]}
                    text="Par SMS"
                  />,
                ],
                "style": Object {
                  "flex": 1,
                  "flexDirection": "row",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "button-1",
                  "nodeType": "function",
                  "props": Object {
                    "align": "center",
                    "disabled": false,
                    "onPress": [Function],
                    "text": "Par Email",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "button-2",
                  "nodeType": "function",
                  "props": Object {
                    "align": "center",
                    "disabled": false,
                    "onPress": [Function],
                    "text": "Par SMS",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Logo />,
            <ReduxForm
              editable={true}
              onRegisterClicked={[Function]}
              onSubmit={[Function]}
            />,
            false,
          ],
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "editable": true,
              "onRegisterClicked": [Function],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <PopupDialog
            dialogAnimation={
              SlideAnimation {
                "animate": 0,
                "animations": Object {
                  "transform": Array [
                    Object {
                      "translateY": 800,
                    },
                  ],
                },
              }
            }
            dialogTitle={
              <DialogTitle
                haveTitleBar={true}
                title="Code d'authentification"
                titleAlign="center"
              />
            }
            height={170}
            width={0.95}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Styled(Text)
                style={
                  Object {
                    "color": "#555",
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
                virtual={undefined}
              >
                Comment souhaiter-vous recevoir votre code d'authentification?
              </Styled(Text)>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <DialogButton
                  align="center"
                  disabled={false}
                  onPress={[Function]}
                  text="Par Email"
                />
                <DialogButton
                  align="center"
                  disabled={false}
                  onPress={[Function]}
                  text="Par SMS"
                />
              </View>
            </View>
          </PopupDialog>,
          <Styled(Content)
            virtual={undefined}
          >
            <Logo />
            <ReduxForm
              editable={true}
              onRegisterClicked={[Function]}
              onSubmit={[Function]}
            />
          </Styled(Content)>,
        ],
        "style": Object {
          "backgroundColor": "#2c3e50",
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Styled(Text)
                style={
                  Object {
                    "color": "#555",
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
                virtual={undefined}
              >
                Comment souhaiter-vous recevoir votre code d'authentification?
              </Styled(Text)>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <DialogButton
                  align="center"
                  disabled={false}
                  onPress={[Function]}
                  text="Par Email"
                />
                <DialogButton
                  align="center"
                  disabled={false}
                  onPress={[Function]}
                  text="Par SMS"
                />
              </View>
            </View>,
            "dialogAnimation": SlideAnimation {
              "animate": 0,
              "animations": Object {
                "transform": Array [
                  Object {
                    "translateY": 800,
                  },
                ],
              },
            },
            "dialogTitle": <DialogTitle
              haveTitleBar={true}
              title="Code d'authentification"
              titleAlign="center"
            />,
            "height": 170,
            "width": 0.95,
          },
          "ref": [Function],
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Styled(Text)
                  style={
                    Object {
                      "color": "#555",
                      "fontSize": 20,
                      "textAlign": "center",
                    }
                  }
                  virtual={undefined}
                >
                  Comment souhaiter-vous recevoir votre code d'authentification?
                </Styled(Text)>,
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <DialogButton
                    align="center"
                    disabled={false}
                    onPress={[Function]}
                    text="Par Email"
                  />
                  <DialogButton
                    align="center"
                    disabled={false}
                    onPress={[Function]}
                    text="Par SMS"
                  />
                </View>,
              ],
              "style": Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Comment souhaiter-vous recevoir votre code d'authentification?",
                  "style": Object {
                    "color": "#555",
                    "fontSize": 20,
                    "textAlign": "center",
                  },
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": "Comment souhaiter-vous recevoir votre code d'authentification?",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <DialogButton
                      align="center"
                      disabled={false}
                      onPress={[Function]}
                      text="Par Email"
                    />,
                    <DialogButton
                      align="center"
                      disabled={false}
                      onPress={[Function]}
                      text="Par SMS"
                    />,
                  ],
                  "style": Object {
                    "flex": 1,
                    "flexDirection": "row",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "button-1",
                    "nodeType": "function",
                    "props": Object {
                      "align": "center",
                      "disabled": false,
                      "onPress": [Function],
                      "text": "Par Email",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": "button-2",
                    "nodeType": "function",
                    "props": Object {
                      "align": "center",
                      "disabled": false,
                      "onPress": [Function],
                      "text": "Par SMS",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Logo />,
              <ReduxForm
                editable={true}
                onRegisterClicked={[Function]}
                onSubmit={[Function]}
              />,
              false,
            ],
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "editable": true,
                "onRegisterClicked": [Function],
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [MockFunction],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
